### ISSUE TITLE
Visual Studio 2010 support

### ISSUE THREAD

**Issue by grahamreeds** [Crystalnix/omaha#27]
I am looking at demoing Omaha (whether vanilla or the crystalnix flavour) and the main stumbling block I have is that we don't have VS 2008 - we have 2010, 2012 and 2013.

I am loathe to go through the same setup process with the CrystalNix flavour just to find it doesn't compile as well.  The Wiki doesn't help as it is a direct copy of the original.

Can I use Visual Studio 2010 with this flavour of Omaha?

**Comment by shashkin** [comment]
The short answer is: No, you have to use Visual Studio 2008
If you don't have one, you might find helpful to follow the other topic of this repo: #26 

At first we tried to use Visual Studio 2010 and I was able to build the project with it after some modifications (see 0206cfaefafc51f1faaa42e6c7f29640d9e40015), but the client did not work as expected. Thats why we decided to use VS2008 then and stated it explicitly in our wiki.

If you or someone else find the way to move this project to the newer version of Visual Studio - pull requests are always welcome :)

**Comment by grahamreeds** [comment]
When you say "Did not work as expected" how long did you put into attempting to fix those issues and what were they in particular?

I am looking at the original and currently have issues with gtest kicking out errors regarding tuple.  It appears that the version of gtest in Omaha is 1.4.0 so I was looking into moving that onto 1.5.0 which was released in April 2010, so the support might be better due to access to a RC version of VS2010.  Failing that a move to 1.6.0 might correct the issue or it might just introduce more.

If I am to invest a large amount of time in this I will simply fork the original and see if I can get it working.  Really, locking the compilable version at 2008 is a non-starter.

**Comment by shashkin** [comment]
Hi @grahamreeds,

When I say "Did not work as expected" I mean I was able to build the project and run it (after 0206cfa).
The client seemed to work, but one of the worker process was failing to exit normally. That completely prevents the subsequent working processes from doing their job and that's definitely not what I'm expected. I assume the problem is somewhere in ATL or in other dependent library included into Visual Studio 2010.

The solution we chose is to switch to Visual Studio 2008 and at this moment we're completely satisfied with this solution in our project. Moving this 4+ years old code to the newer VS version is not the priority for us.
